pkg/volume/volume.go

```diff
@@ -65,6 +66,12 @@ type Recycler interface {
 	Recycle() error
 }
 
+// Create adds a new resource in the storage provider and creates a PersistentVolume for the new resource.
+// Calls to Create should block until complete.
+type Creater interface {
+	Create() (*api.PersistentVolume, error)
```

Comment:
Yeah, I think that if we're retooling the controllers this makes more sense to be lifted out of the plugin.

Also, we've done sort of a poor job of defining exactly what logic belongs in the the GCE volume code vs what belongs in the GCE cloud provider code.  More and more I am feeling that the GCE-API stuff should be in the cloud provider - the GCE specific provider, not as a generic abstraction.  That module becomes our "one true place" to interface with GCE APIs.  The GCEPD volume uses that to get PDs attached and mounted.  (I mean eventually :)